### PR TITLE
add numeric size support to Avatar's badge + unit-tests

### PR DESCRIPTION
**Pull request by mendyEdri**
Added numeric value support for Avatar's badge size.

### CODE REVIEW

File: src/components/avatar/__tests__/index.spec.js [hunk 1]
```diff
@@ -58,4 +59,40 @@ describe('Avatar Badge', () => {
...
+    it('should return 876 as the size number given', () => {
+      const uut = new Avatar({badgeProps: {size: 876}});
+      expect(uut.getBadgeSize()).toEqual(876);
+    });
+
+    it('should return 0 as the given size number', () => {
+      const uut = new Avatar({badgeProps: {size: 0}});
+      expect(uut.getBadgeSize()).toEqual(0);
+    });
+
```
Comment (ArnonZ) [inline src/components/avatar/__tests__/index.spec.js:78]: Not the end of the world obviously but having 3 tests testing numeric input might be 2 tests too much 😸 .
If one of these tests fail, all of them will fail for the same reason.
For your consideration...

File: src/components/avatar/index.tsx [hunk 2]
```diff
@@ -222,14 +222,21 @@ class Avatar extends PureComponent<AvatarPropTypes> {
     return _.get(this.props, 'badgeProps.backgroundColor') || statusColor || onlineColor;
   }
 
-  getBadgeSize = () => _.get(this.props, 'badgeProps.size', DEFAULT_BADGE_SIZE);
+  getBadgeSize = () => {
+    const badgeSize = _.get(this.props, 'badgeProps.size', DEFAULT_BADGE_SIZE);
+    
+    if (_.isString(badgeSize)) {
+      return BADGE_SIZES[badgeSize];
```
Comment (ArnonZ) [inline src/components/avatar/index.tsx:229]: When you split the logic between enum and numeric size you eliminated the fallback of `DEFAULT_BADGE_SIZE` for string-based sizes (currently returns undefined if sending a bad enum). Consider adding `DEFAULT_BADGE_SIZE` fallback if supplied enum size isn't included in `BADGE_SIZES` - to maintain old fallback logic.

Comment (mendyEdri) [reply src/components/avatar/index.tsx]: Right, added.

File: src/components/avatar/__tests__/index.spec.js (hunk 1 above)
Comment (mendyEdri) [reply src/components/avatar/__tests__/index.spec.js:73]: right, keeping two for zero/undefined check + some number

File: src/components/avatar/__tests__/index.spec.js [hunk 3]
```diff
@@ -58,4 +59,35 @@ describe('Avatar Badge', () => {
       expect(uut.renderBadge()).toEqual(undefined);
     });
   });
+
+  describe('badgeProps.size, supports enum or number', () => {
+    it('should return 10 as the size number given', () => {
```
Comment (ArnonZ) [inline src/components/avatar/__tests__/index.spec.js:64]:
```suggestion
    it('should return 99 as the size number given', () => {
```

**Pull request merged**